pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +354,132 @@ type BGPPeer struct {
+	// HopLimit is the IPv6 Hop Limit.
+	HopLimit *int32 `json:"hopLimit,omitempty" yaml:"hopLimit,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMP *ICMPEchoRequestHeader `json:"icmp,omitempty" yaml:"icmp,omitempty"`
+	UDP  *UDPHeader             `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP  *TCPHeader             `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	IPv6Header      *IPv6Header     `json:"ipv6Header,omitempty"`
+	TransportHeader TransportHeader `json:"transportHeader"`
+}
+
+// PacketCaptureFirstNConfig contains the config for the FirstN type capture. The only supported parameter is
+// `Number` at the moment, meaning capturing the first specified number of packets in a flow.
+type PacketCaptureFirstNConfig struct {
+	Number int32 `json:"number"`
+}
+
+const DefaultPacketCaptureTimeout uint16 = 60
+
+type PacketCapturePhase string
+
+const (
+	PacketCaptureRunning   PacketCapturePhase = "Running"
+	PacketCaptureSucceeded PacketCapturePhase = "Succeeded"
+	PacketCaptureFailed    PacketCapturePhase = "Failed"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCaptureList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PacketCapture `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCapture struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              PacketCaptureSpec   `json:"spec"`
+	Status            PacketCaptureStatus `json:"status"`
+}
+
+type CaptureConfig struct {
+	FirstN *PacketCaptureFirstNConfig `json:"firstN,omitempty"`
+}
+
+type PacketCaptureSpec struct {
+	Timeout       *uint16       `json:"timeout,omitempty"`
+	CaptureConfig CaptureConfig `json:"captureConfig"`
+	Source        Source        `json:"source"`
+	Destination   Destination   `json:"destination"`
+	Packet        *Packet       `json:"packet,omitempty"`
+	// FileServer specifies the sftp url config for the fileServer. Captured packets will be uploaded to this server.
+	FileServer BundleFileServer `json:"fileServer"`
```

Comment:
I feel like we should have the option to write the pcap to a local file as well, just for ease of use / testing
Not everyone will want to have an ftp server for this

Reply:
So do you think keep the files in contains is enough?  Since this is not a cli, it would be odd to export the files to a location on the host.

If keep it in the container, filepath + container name should be ok?

Reply:
> If keep it in the container, filepath + container name should be ok

Just filepath I think, the container will always be `antrea-agent`? If users want the file to be on the Node, they can provide a path that is mounted from the host, such as `/var/log/antrea`.

But there would need to be a way for users to figure out which antrea-agent Pod has the capture file, so that would need to be included in the Status. Are packets captured at the source or the destination or both?

Reply:
Both could happen, in the source or destination.  pod name can be exposed in the status. Just need to think about the value format for the filepath field.

Reply:
I think my original point still applies. Always requiring users to provide an SFTP server in order to do a packet capture is not ideal. That being said, if I am the only one who feels strongly about this, we can always address it in the future if needed.

Reply:
Agreed here. I have the following plans:

1. make `FileServer` field optional.
2. Change the `status.packetsFileName` to `status.packetsFilePath` and it's value will be like `antrea-agent-xxxx:/tmp/xxx.pcap

Does this sounds good?

Reply:
see: https://github.com/antrea-io/antrea/pull/6257/commits/939fe064c42d1498353bfd98d8ecc2ba07e38181